Mathlib/RingTheory/RingInvo.lean

```diff
@@ -0,0 +1,95 @@
+/-
+Copyright (c) 2018 Andreas Swerdlow. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Andreas Swerdlow, Kenny Lau
+
+! This file was ported from Lean 3 source module ring_theory.ring_invo
+! leanprover-community/mathlib commit 09597669f02422ed388036273d8848119699c22f
+! Please do not edit these lines, except to modify the commit id
+! if you have ported upstream changes.
+-/
+import Mathlib.Algebra.Ring.Equiv
+import Mathlib.Algebra.Ring.Opposite
+
+/-!
+# Ring involutions
+
+This file defines a ring involution as a structure extending `R ≃+* Rᵐᵒᵖ`,
+with the additional fact `f.involution : (f (f x).unop).unop = x`.
+
+## Notations
+
+We provide a coercion to a function `R → Rᵐᵒᵖ`.
+
+## References
+
+* <https://en.wikipedia.org/wiki/Involution_(mathematics)#Ring_theory>
+
+## Tags
+
+Ring involution
+-/
+
+
+variable (R : Type _)
+
+/-- A ring involution -/
+structure RingInvo [Semiring R] extends R ≃+* Rᵐᵒᵖ where
+/-- The requirement that the ring homomorphism is its own inverse -/
+  involution' : ∀ x, (toFun (toFun x).unop).unop = x
+#align ring_invo RingInvo
+
+namespace RingInvo
+
+variable {R} [Semiring R]
+
+/-- Construct a ring involution from a ring homomorphism. -/
+def mk' (f : R →+* Rᵐᵒᵖ) (involution : ∀ r, (f (f r).unop).unop = r) : RingInvo R :=
+  { f with
+    invFun := fun r => (f r.unop).unop
+    left_inv := fun r => involution r
+    right_inv := fun _ => MulOpposite.unop_injective <| involution _
+    involution' := involution }
+#align ring_invo.mk' RingInvo.mk'
+
+instance : CoeFun (RingInvo R) fun _ => R → Rᵐᵒᵖ :=
+  ⟨fun f => f.toRingEquiv.toFun⟩
+
+#noalign ring_invo.to_fun_eq_coe
+
+@[simp]
+theorem involution (f : RingInvo R) (x : R) : (f (f x).unop).unop = x :=
```

Comment:
```suggestion
theorem involution (f : RingInvo R) (x : R) : (f.toRingEquiv  (f.toRingEquiv  x).unop).unop = x :=
```
Maybe this is how to handle the lint error here? I'm really not sure.

Reply:
What lint error do you have?

Reply:
I think there is no issue anymore, right?